### ISSUE TITLE
"Steamwebhelper is not responding" shortly followed by "Cannot read properties of undefined (reading 'GetPlayer')"

### ISSUE THREAD

**Issue by TheArcaneBrony** [ValveSoftware/steam-for-linux#11079]
#### Your system information

* Steam client version (build number or date): Unknown, presuming latest (`Startup - updater built Jun 20 2024 17:13:49`)
* Distribution (e.g. Ubuntu): NixOS 24.11
* Opted into Steam client beta?: No
* Have you checked for system updates?: Yes
* Steam Logs: [steam-logs.tar.gz](https://github.com/user-attachments/files/16131431/steam-logs.tar.gz)
* GPU: AMD
* Desktop environment: Sway (Wayland/WLRoots)

#### Please describe your issue in as much detail as possible:
Issue created after seeing multiple similar reports in #10431, but no NFS/bindmount circumstances are present on my system.
None of the restart options seem to have any effect (not even seeing steam restart at all).

#### Steps for reproducing this issue:

1. Launch steam
2. Observe "Steamwebhelper is not responding" pop up
3. Observe JS error showing up a few seconds later

**Comment by TheArcaneBrony** [comment]
For some unknown reason, rebooting seems to have fixed my issue.
I wonder if it was some internal service launch error that caused the GetPlayer API to not be available? Didn't notice anything odd in stdout.
Could the failure be triggered by having a slow process start (eg. waiting for dbus/desktop portals/...)?

With the client in the "JS error" state, i still saw steam consuming network resources to update games.

**Comment by tsukasa1234567** [comment]
Not sure if it helps, but I had that error for a few days.  Not good for a holiday weekend, but I it seemed to get fixed for me by updating to kernel 6.9.7, it seems a lot of people have had issues in the 6.9 series but I didn't see anything glaring in the changelog to figure anything out.

**Comment by ms178** [comment]
Could be due to a recent xwayland regression, see: https://gitlab.freedesktop.org/xorg/xserver/-/issues/1696 with the WIP fix here: https://gitlab.freedesktop.org/xorg/xserver/-/merge_requests/1574

**Comment by tsukasa1234567** [comment]
I'm running MATE, pure xorg, seems a lot better for gaming still.  No weird tabbing issues or anything like that.  Totally appreciate the suggestion though.

**Comment by schpitz** [comment]
Still happens to me as well. The exact same error message.
I've tried to uninstall and reinstall, restart it with GPU acceleration disabled, with Browser sandboxing disabled, and nothing works. It worked fine only few days ago.
I have Win10, AMD 6600XT, drivers 24.4.1

**Comment by sketchbuch** [comment]
Same here,Ubuntu 24.04. Was working to day until my system updated.

**Comment by tsukasa1234567** [comment]
I miss the old days before pressure-vessel where you could either pick to use steams bundled libs, or use the systems libs at your own risk.   Some sort of chroot jail or similiar seems less prone to all these error's made by remaking flatpak and then trying to have it figure out mixing host and steam libs on the fly for every combination of system out there.

The only thing i've found that helps is to install vmtouch
then cd ~/.local/share/Steam/ubuntu12_64
vmtouch -L ./
wait for it to finish, uses about 1 GB of ram
launch steam, it'll crash hard, but leave the steamwebhelper crash window up, then you'll get the black page of cannot read properties. then pick to close steam in the steamwebhelper crash window.  then restart steam, it generally works for me that 2nd load, but sometimes it takes 3-20.....which is less then the 100+ I've tried reloading in a row without vmtouch

Hope that helps.

@schpitz totally blown away that your getting the error on Win10.  Welcome to the club

**Comment by Fell** [comment]
I have the same issue. It might be related to [9588](https://github.com/ValveSoftware/steam-for-linux/issues/9588). Uninstalling any of the `xdg-desktop-portal` implementations is not an option for me, because I depend on it for Lutris and Firefox' system theme integrations.

However, on my system, Steam always launches correctly after choosing any of the restart options. Only the very first start of the day fails with the described behavior.

Operating System: Arch Linux 
KDE Plasma Version: 6.1.2
KDE Frameworks Version: 6.3.0
Qt Version: 6.7.2
Kernel Version: 6.9.8-arch1-1.1 (64-bit)
Graphics Platform: Wayland
Processors: 24 × AMD Ryzen 9 3900X 12-Core Processor
Memory: 31.3 GiB of RAM
Graphics Processor: NVIDIA GeForce RTX 2080 Ti/PCIe/SSE2

**Comment by AngryPenguinPL** [comment]
It looks like we have a similar error.

After first running the steam installer during the unpacking process of the "Steam Linux runtime container" an error appears.

Clicking restart repeats the error. The only way to get around this is to close the installer window and kill the steam process and then restart it - now steam displays the login window.

Quick info:
OS OpenMandriva Cooker/Rolling
DE: Plasma5, Plasma6 x11/Wayland, GNOME Wayland/X11

I have collected some logs and core dumps, I am posting them below
![Screenshot_20240717_143545](https://github.com/user-attachments/assets/2726b6a9-e28e-482c-b84d-88b3b6f4bbf5)

log running steam installed in terminal: [steam-install.txt](https://github.com/user-attachments/files/16267088/steam-install.txt)

journallog collected during first steam launching:  [journal.txt](https://github.com/user-attachments/files/16267092/journal.txt)

Dumps logs: 
[assert_20240717143459_30.dmp.txt](https://github.com/user-attachments/files/16267230/assert_20240717143459_30.dmp.txt)

[assert_20240717143529_34.dmp.txt](https://github.com/user-attachments/files/16267243/assert_20240717143529_34.dmp.txt)

[assert_20240717143659_29.dmp.txt](https://github.com/user-attachments/files/16267245/assert_20240717143659_29.dmp.txt)

**Comment by smcv** [comment]
@AngryPenguinPL, please open a separate issue and attach the logs from `~/.steam/steam/logs`. You are getting a similar, but different, issue: you can tell it's different because you do not see the error message "Cannot read properties of undefined (reading 'GetPlayer')".

The logs from `~/.steam/steam/logs` are required for any `steamwebhelper` issue. Steam's command-line output does not contain enough information to diagnose `steamwebhelper` crashing or failing to start.

**Comment by edwoollard** [comment]
I just started experiencing this exact issue today. I was playing this morning without issue, turned off my PC and then came back to my PC just now and Steam now won't boot.

**Comment by shrubbo** [comment]
Same here, been like this a few days now.  I get a window with the steam logo in a spinning wheel never completing.

**Comment by FatalCatharsis** [comment]
Got this today attempting to start steam. I killed steam client webhelper and attempted to restart and am still getting the error after it spins for a bit. Any workaround for this?

**Comment by tsukasa1234567** [comment]
Welcome to the party, been dealing with this since January.  Seems related mainly to pressure-vessel not working for anything but ext4.  The only workaround that's currently working for me is make a zram/ext4 at ~/.local/share/Steam/ubuntu12_64
then it crashes the first load, I let it finish and settle activity, then close it and relaunch, usually works on the 2nd load.  Does waste about 1.5 gigs of ram for steam, and another 1.5 for each version of proton you use to play games.  Valve has always had very few employees so it's not a priority for anyone, they usually just act like normal forum moderators and tell ppl they submitted incorrectly or whatever.  I still have hope but 8 months is a long time.  Maybe their should be a bug bounty website or something for steam.

**Comment by sketchbuch** [comment]
My fix was to just uninstall the flatpak version and knstall the deb

**Comment by rubynt** [comment]
it somehow works on gnome wayland but not for xorg.

**Comment by WGH-** [comment]
Do people that have Steam startup problems also have problems starting other applications? 

 I had ridiculously slow startup of many other apps, like `pavucontrol`, or Firefox, which I fixed by removing `xdg-desktop-portal-gtk`, which coincidentally also fixed Steam startup problems.

**Comment by chobert2** [comment]
> Do people that have Steam startup problems also have problems starting other applications?
> 
> I had ridiculously slow startup of many other apps, like `pavucontrol`, or Firefox, which I fixed by removing `xdg-desktop-portal-gtk`, which coincidentally also fixed Steam startup problems.

This worked like a charm.

**Comment by Guiorgy** [comment]
I'm on Windows 10 and seeing the following:

![Cannot read properties of undefined (reading 'GetPlayer') banner at the top of the Steam client](https://github.com/user-attachments/assets/2cc63acb-04e2-4e11-89ee-a1788ed4a888)

So, probably not something isolated to Linux.

**Comment by B14z3DH** [comment]
ok so it kinda fixed, but only after entering big screen mode....

**Comment by nefariouslegion** [comment]
I started having this same issue a couple days ago. Weirdly though for me that only happens on my first attempt to launch steam after booting my computer. If I exit steam and retry it always launches successfully on the 2nd try.

I'm on CachyOS and usually try to use Steam (Native) 

I see @WGH-  reporting that removing xdg-desktop-portal-gtk fixes the issue (I too have slow startup times for Firefox on first boot). But I'm fairly new to Linux and I'm unsure what that is or what it does. So idk if I should remove it or if it might break other things by removing it. Any guidance would be appreciated.

**Comment by ipr** [comment]
possibly same problem that I'm having: #10813

**Comment by ipr** [comment]
I think there was mentioned at a blog-post that problem exists due to a bug in glibc, which has been there for at least a decade according to different posts (setenv not being thread-safe). Also there was apparently some working being done for it but it hasn't been released? (Not included in most recent glibc version.)
